File: docs/src/test/java/com/example/javadsl/DropwizardExample.java
package com.example.javadsl;

import akka.actor.ActorSystem;
import akka.stream.ActorMaterializer;
import akka.stream.Materializer;
import akka.stream.checkpoint.CheckpointBackend;
import akka.stream.checkpoint.DropwizardBackend;
import akka.stream.checkpoint.javadsl.Checkpoint;
import akka.stream.javadsl.Source;
import com.codahale.metrics.MetricRegistry;

public class DropwizardExample {

    public static void main(String[] args) {
        final ActorSystem system        = ActorSystem.create("DropwizardExample");
        final Materializer materializer = ActorMaterializer.create(system);

        // #dropwizard
        final MetricRegistry metricRegistry = new MetricRegistry();
        final CheckpointBackend backend     = DropwizardBackend.fromRegistry(metricRegistry);

        Source.range(1, 100)
            .via(Checkpoint.create("produced", backend))
            .filter(x -> x % 2 == 0)
            .via(Checkpoint.create("filtered", backend))
            .runForeach(System.out::println, materializer);
        // #dropwizard
    }
}

File: docs/src/test/scala/com/example/scaladsl/DropwizardExample.scala
package com.example.scaladsl

import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import akka.stream.checkpoint.scaladsl.Checkpoint
import akka.stream.scaladsl.Source
import com.codahale.metrics.MetricRegistry

object DropwizardExample extends App {
  implicit val system       = ActorSystem.create("DropwizardScalaExample")
  implicit val materializer = ActorMaterializer.create(system)

  // #dropwizard
  implicit val metricRegistry = new MetricRegistry()

  import akka.stream.checkpoint.DropwizardBackend._

  Source(1 to 100)
    .via(Checkpoint("produced"))
    .filter((x: Int) ⇒ x % 2 == 0)
    .via(Checkpoint("filtered"))
    .runForeach(println)
  // #dropwizard
}
